web_dashboard_tile/models/tile_tile.py

```diff
@@ -1,155 +1,127 @@
-        return [('id', 'in', ids)]
+    def _get_eval_context(self):
+        def _context_today():
+            return fields.Date.from_string(fields.Date.context_today(self))
+        context = self.env.context.copy()
+        context.update({
+            'time': time,
+            'datetime': datetime,
+            'relativedelta': relativedelta,
+            'context_today': _context_today,
+            'current_date': time.strftime('%Y-%m-%d'),
+        })
+        return context
 
     # Column Section
     name = fields.Char(required=True)
-    model_id = fields.Many2one(
-        comodel_name='ir.model', string='Model', required=True)
-    user_id = fields.Many2one(
-        comodel_name='res.users', string='User')
+    sequence = fields.Integer(default=0, required=True)
+    user_id = fields.Many2one('res.users', 'User')
+    background_color = fields.Char(default='#0E6C7E', oldname='color')
+    font_color = fields.Char(default='#FFFFFF')
+
+    model_id = fields.Many2one('ir.model', 'Model', required=True)
     domain = fields.Text(default='[]')
-    action_id = fields.Many2one(
-        comodel_name='ir.actions.act_window', string='Action')
-    count = fields.Integer(compute='_get_tile_info')
-    computed_value = fields.Float(compute='_get_tile_info')
-    helper = fields.Char(compute='_get_tile_info')
-    field_function = fields.Selection(selection=[
+    action_id = fields.Many2one('ir.actions.act_window', 'Action')
+
+    count = fields.Integer(compute='_compute_data')
+    computed_value = fields.Float(compute='_compute_data')
+
+    field_function = fields.Selection([
         ('min', 'Minimum'),
         ('max', 'Maximum'),
         ('sum', 'Sum'),
         ('avg', 'Average'),
         ('median', 'Median'),
         ], string='Function')
     field_id = fields.Many2one(
-        comodel_name='ir.model.fields', string='Field',
+        'ir.model.fields',
+        string='Field',
         domain="[('model_id', '=', model_id),"
         " ('ttype', 'in', ['float', 'int'])]")
-    active = fields.Boolean(
-        compute='_get_tile_info', readonly=True, search='_search_active')
-    background_color = fields.Char(default='#0E6C7E', oldname='color')
-    font_color = fields.Char(default='#FFFFFF')
-    sequence = fields.Integer(default=0, required=True)
+    helper = fields.Char(compute='_compute_function_helper')
+
+    @api.one
+    def _compute_data(self):
+        self.count = 0
+        self.computed_value = 0
+        # Compute count item
+        model = self.env[self.model_id.model]
+        eval_context = self._get_eval_context()
+        self.count = model.search_count(eval(self.domain, eval_context))
+        # Compute datas for field_id depending of field_function
+        if self.field_function and self.field_id and self.count != 0:
+            records = model.search(eval(self.domain, eval_context))
+            vals = [x[self.field_id.name] for x in records]
+            if self.field_function == 'min':
```

Comment:
Something more Pythonic that would accomplish the same w/o the overhead of all these if statements:

```
function_map = {
    'min': lambda x: min(x),
    'avg': lambda x: sum(x) / len(x),
    ...
}
try:
    self.computed_value = function_map[self.field_function](vals)
except KeyError:
    pass
```